### PR TITLE
feat(init-1d): #MA-998 start init 1D form display

### PR DESCRIPTION
**Pull request by flmartineau**
## Describe your changes

- Add popup for Viescolaire initialisation in Présences

CSS : https://github.com/CGI-OPEN-ENT-NG/entcore-css-lib/pull/52

Documentation : https://confluence.support-ent.fr/pages/viewpage.action?pageId=158075429

## Checklist tests

- check if the popup displays only if the structure has not been initialized yet
- check if the API call for initialisation is correct

## Issue ticket number and link

https://jira.support-ent.fr/browse/MA-998

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: presences/src/main/resources/public/ts/directives/init-lightbox/init-lightbox.directive.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                        element: ng.IAugmentedJQuery,
+                        attrs: ng.IAttributes,
+                        vm: IViewModel) {
+
+
+        }
+    }
+}
+
+export const initLightbox = ng.directive('initLightbox', directive)
```
Comment (JinDouang) [inline presences/src/main/resources/public/ts/directives/init-lightbox/init-lightbox.directive.ts]:
```suggestion
export const initLightbox = ng.directive('initLightbox', directive);
```

File: presences/src/main/resources/public/ts/services/initialization.service.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            .then((res: AxiosResponse) => res.data);
+    },
+
+    getViescoInitStatus: async (structureId: string): Promise<IInitStatusResponse> => {
+        return http.get(`/viescolaire/structures/${structureId}/initialization`)
+            .then((res: AxiosResponse) => res.data);
+    },
+
+    initPresences: async (structureId: string, initType: INIT_TYPE): Promise<void> => {
+        await http.post(`/presences/initialization/structures/${structureId}`, {init_type: initType});
```
Comment (JinDouang) [inline presences/src/main/resources/public/ts/services/initialization.service.ts]: Pourquoi pas simplement return le http.post ?

Si un composant souhaite utiliser le service, pas grave s'il ne se sert pas de la réponse

mais si un autre souhaite l'utiliser mais qui veut dans son composant vérifier le status ou autre....

File: presences/src/main/resources/public/ts/services/initialization.service.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        return;
+    },
+
+    initViesco: async (structureId: string, initType: INIT_TYPE, initForm: InitForm): Promise<void> => {
+        initForm.initType = initType;
+        await http.post(`/viescolaire/structures/${structureId}/initialize`, initForm.toJSON());
```
Comment (JinDouang) [inline presences/src/main/resources/public/ts/services/initialization.service.ts]: idem

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: presences/src/main/resources/public/ts/directives/init-lightbox/init-lightbox.html [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                        <label for="system-select-other">
+                            <i18n>presences.init.1d.form.holidays.other.system</i18n>
+                        </label>
+                    </div>
+                </div>
+                <div class="init-1d-lightbox-body-holidays-content-zone-select">
+                    <span><i18n>presences.init.1d.form.holidays.detected.zone</i18n> :</span>
+                    <div class="init-1d-lightbox-body-holidays-content-zone-select-zones">
+                        <div class="md-radio md-radio-inline">
+                            <input type="radio" ng-model="vm.form.holidays.zone"
```
Comment (Sofianebernoussi) [inline presences/src/main/resources/public/ts/directives/init-lightbox/init-lightbox.html:134]: boucle et/ou directive peut être pour les zone?

File: presences/src/main/resources/public/ts/sniplets/presences-manage.ts [hunk 5]
```diff
@@ -47,25 +47,25 @@ function safeApply() {
...
 
 function fetchInitializationStatus(): void {
-    http.get(`/presences/initialization/structures/${window.model.vieScolaire.structure.id}`)
-        .then(({data}) => {
-            if ('initialized' in data) vm.initialized = data.initialized;
-            else vm.initialized = false;
-            vm.safeApply();
-        })
-        .catch(err => console.error('Failed to retrieve structure initialization status', err));
+    initService.getPresencesInitStatus(window.model.vieScolaire.structure.id).then((r: IInitStatusResponse) => {
```
Comment (Sofianebernoussi) [inline presences/src/main/resources/public/ts/sniplets/presences-manage.ts]: change le r par un truc plus parlant

**Review by Sofianebernoussi — APPROVED** [review]

**Review by BastienBenard — APPROVED** [review]

File: presences/src/main/resources/public/ts/directives/init-lightbox/init-lightbox.html (hunk 4 above)
Comment (flmartineau) [reply presences/src/main/resources/public/ts/directives/init-lightbox/init-lightbox.html:134]: Il y aura toujours 3 zones, pas plus, donc une boucle n'est pas nécessaire.

**Pull request merged**